engine/crates/composition/src/compose/object.rs

```diff
@@ -216,3 +216,28 @@ fn collect_overrides(fields: &[FieldWalker<'_>], ctx: &mut Context<'_>) -> Vec<f
 
     overrides
 }
+
+pub(crate) fn validate_shareable_object_fields_match(
+    definitions: &[DefinitionWalker<'_>],
+    ctx: &mut ComposeContext<'_>,
+) {
+    let all_fields: BTreeSet<StringId> = definitions
+        .iter()
+        .flat_map(|def| def.fields())
+        .filter(|field| !field.directives().inaccessible())
```

Comment:
This is wrong: if _any_ of the definitions of the field is inaccessible, that's enough for the field to be inaccessible. There should be a test for that.